crates/typst/src/text/font/exceptions.rs

```diff
@@ -0,0 +1,223 @@
+use serde::Deserialize;
+
+use super::{FontStretch, FontStyle, FontWeight};
+
+#[derive(Debug, Default, Deserialize)]
+pub struct Exception {
+    pub family: Option<&'static str>,
+    pub style: Option<FontStyle>,
+    pub weight: Option<FontWeight>,
+    pub stretch: Option<FontStretch>,
+}
+
+pub fn find_exception(postscript_name: &str) -> Option<&'static Exception> {
+    EXCEPTION_MAP.get(postscript_name)
+}
+
+/// A map which keys are PostScript name and values are override entries.
+static EXCEPTION_MAP: phf::Map<&'static str, Exception> = phf::phf_map! {
+    "NewCM08-Book" => Exception {
+        family: Some("New Computer Modern 08"),
+        weight: Some(FontWeight(450)),
+        style: None,
+        stretch: None,
+    },
+    "NewCM08-BookItalic" => Exception {
+        family: Some("New Computer Modern 08"),
+        weight: Some(FontWeight(450)),
+        style: None,
+        stretch: None,
+    },
+    "NewCM08-Italic" => Exception {
+        family: Some("New Computer Modern 08"),
+        weight: None,
+        style: None,
+        stretch: None,
+    },
+    "NewCM08-Regular" => Exception {
+        family: Some("New Computer Modern 08"),
+        weight: None,
+        style: None,
+        stretch: None,
+    },
+    "NewCM10-Bold" => Exception {
+        family: Some("New Computer Modern"),
+        weight: None,
+        style: None,
+        stretch: None,
+    },
+    "NewCM10-BoldItalic" => Exception {
+        family: Some("New Computer Modern"),
+        weight: None,
+        style: None,
+        stretch: None,
+    },
+    "NewCM10-Book" => Exception {
+        family: Some("New Computer Modern"),
+        weight: Some(FontWeight(450)),
+        style: None,
+        stretch: None,
+    },
+    "NewCM10-BookItalic" => Exception {
+        family: Some("New Computer Modern"),
+        weight: Some(FontWeight(450)),
+        style: None,
+        stretch: None,
+    },
+    "NewCM10-Italic" => Exception {
+        family: Some("New Computer Modern"),
+        weight: None,
+        style: None,
```

Comment:
Why does this font italic, but the style/weight is empty?

Reply:
Because Typst detect the italic style just fine, so we do not specify exception here.